model/DataObject.php

```diff
@@ -1744,44 +1745,79 @@ public function getManyManyComponents($componentName, $filter = "", $sort = "",
 	}
 
 	/**
+	 * @deprecated 4.0 Method has been renamed to hasOne()
+	 * @param string $component
+	 * @return array|null
+	 */
+	public function has_one($component = null) {
+		Deprecation::notice('4.0', 'Method has been renamed to hasOne()');
```

Comment:
... and I’ve shortened them again. When thrown with `Deprecation::SCOPE_METHOD` (default 2nd arg), it already includes the name of the original method. So the output of the above is:

`DataObject:: has_one is deprecated. Method has been renamed to hasOne()`